### PR TITLE
Misc. Kconfig and testcase fixes and enhancements

### PR DESCRIPTION
**Pull request by nashif**
This all is part of an attemp to cleanup tests and samples and increase
coverage of tests.

### CODE REVIEW

File: tests/crypto/mbedtls/prj.conf [hunk 1]
```diff
@@ -1,6 +1,4 @@
-# nothing yet
 CONFIG_MAIN_STACK_SIZE=4096
-CONFIG_ARC_INIT=n
```
Comment (galak) [inline tests/crypto/mbedtls/prj.conf]: What does the ARC_INIT have to do with DEBUG?

File: tests/crypto/mbedtls/testcase.yaml [hunk 2]
```diff
@@ -3,3 +3,4 @@ tests:
         min_ram: 32
         tags: crypto mbedtls
         timeout: 200
+        filter: not CONFIG_DEBUG
```
Comment (galak) [inline tests/crypto/mbedtls/testcase.yaml]: should we have a bug opened for the fact that mbedtls has issues with DEBUG (not sure what those issues are)?

**Review by galak — CHANGES_REQUESTED** [review]
The spi move seems to have one too many spi/spi/ dirs.  We should probably either rename the test or reduce a level of dir.

File: tests/crypto/mbedtls/testcase.yaml (hunk 2 above)
Comment (nashif) [inline tests/crypto/mbedtls/testcase.yaml]: the issues is related to our warnings turning into errors, it is not necessarily an issue in mbedTLS, so opening a bug here will just make it stay open forever, not sure it is worth the trouble.

**Comment by nashif** [comment]
updated

File: boards/arm/frdm_k64f/frdm_k64f.yaml [hunk 3]
```diff
@@ -5,3 +5,5 @@ arch: arm
 toolchain:
   - zephyr
   - gccarmemb
+supported:
+  - net
```
Comment (galak) [inline boards/arm/frdm_k64f/frdm_k64f.yaml]: we should make this be ethernet not net

**Comment by galak** [comment]
The kw41z board should have ieee802154 marked in its HW support

File: include/misc/stack.h [hunk 4]
```diff
@@ -26,7 +26,7 @@ static inline size_t stack_unused_space_get(const char *stack, size_t size)
 	 * (or configurable direction) is added this check should be confirmed
 	 * that correct Kconfig option is used.
 	 */
-#if defined(CONFIG_STACK_GROWS_UP)
+#if defined(STACK_GROWS_UP)
```
Comment (galak) [inline include/misc/stack.h]: where is STACK_GROWS_UP defined anywhere?

File: tests/crypto/mbedtls/src/mbedtls_test.c [hunk 5]
```diff
@@ -431,5 +435,6 @@ int main(void)
 	}
 
 	while (1) {
-	};
+	}
+	;
```
Comment (galak) [inline tests/crypto/mbedtls/src/mbedtls_test.c]: this seems odd, why do we even have the ;

**Review by galak — CHANGES_REQUESTED** [review]
some changes, needed.  Created PR #693 to pull the commits that are ready to merge into, so only those commits that need fixing up will need to get reviewed.

File: include/misc/stack.h (hunk 4 above)
Comment (nashif) [inline include/misc/stack.h]: read the comment on top of this line

Comment (galak) [inline include/misc/stack.h]: Ah

File: boards/arm/frdm_kw41z/frdm_kw41z.yaml [hunk 6]
```diff
@@ -5,3 +5,5 @@ arch: arm
 toolchain:
   - zephyr
   - gccarmemb
+supported:
+  - ieee802154
```
Comment (galak) [inline boards/arm/frdm_kw41z/frdm_kw41z.yaml]: should we put ble here, even if the sw doesn't support it?

File: boards/x86/qemu_x86/qemu_x86.yaml [hunk 7]
```diff
@@ -6,3 +6,4 @@ toolchain:
   - zephyr
 supported:
   - pci
+  - net
```
Comment (galak) [inline boards/x86/qemu_x86/qemu_x86.yaml]: what is net suppose to mean here?

File: boards/x86/qemu_x86/qemu_x86_iamcu.yaml [hunk 8]
```diff
@@ -6,5 +6,6 @@ toolchain:
   - zephyr
 supported:
   - pci
+  - net
```
Comment (galak) [inline boards/x86/qemu_x86/qemu_x86_iamcu.yaml]: what is net suppose to mean here?

**Review by galak — CHANGES_REQUESTED** [review]
what is 'net' suppose to be mean for qemu?

**Comment by galak** [comment]
Also, don't we have networking support on qemu_cortex_m3?

File: boards/x86/qemu_x86/qemu_x86.yaml (hunk 7 above)
Comment (nashif) [inline boards/x86/qemu_x86/qemu_x86.yaml]: networking in general, this is not not tied to a HW device.

**Comment by nashif** [comment]
> Also, don't we have networking support on qemu_cortex_m3?

via SLIP yes, but does it make sense to test on all QEMUs? We need to pick one and do the test via SLIP on one. This is pretty much the same and we are not testing HW features here, we are testing the IP stack. Dont care which we use, qemu_x86 was used sofar.

**Comment by galak** [comment]
Lets see if @pfalcon has a feeling on testing net with qemu.  I know he's dealt with this and want to make sure the qemu works reasonable for it on any/all platforms that we support.

**Comment by pfalcon** [comment]
Well, ideally, we'd test both qemu_x86 and qemu_cortex_m3, to make sure there're no regressions in qemu_cortex_m3, because nobody likely uses it for testing ;-).

Otherwise, to run Zephyr sample with SLIP enabled in QEMU, there should be running daemons for SLIP/TAP conversion - at the very least. So, we can't run such tests using normal sanitycheck processing (and indeed, there're nos such *tests*), and fairly speaking, I don't know how to handle that at all (there can be various ideas of course). But that matter needs to be addressed somehow, as the net code is already complex, and complex refactors keep being applied to it (I just complained on that at https://github.com/zephyrproject-rtos/zephyr/pull/691#issuecomment-313415170), and there're no good way to regression-test them.

**Review by galak — APPROVED** [review]

**Pull request merged**